spec/acceptance/functions/hosts_with_pe_profile_spec.rb

```diff
@@ -0,0 +1,12 @@
+require 'spec_helper_acceptance'
+
+describe 'hosts_with_pe_profile function' do
+  it 'returns 127.0.0.1' do
+    pp = <<-MANIFEST
+       $result=puppet_metrics_collector::hosts_with_pe_profile('non_existant')
+       notice "Received=$result"
+       MANIFEST
+    output = apply_manifest(pp).stdout
+    expect(output).to match(%r{127.0.0.1})
+  end
```

Comment:
Can we add a test case for the function with a matching result? To do that we would need to run a shell to do `puppet config set storeconfigs true --section user` and then query the function for the `master` profile. It should return an array containing `host_inventory['fqdn']`